src/bindings/python/flux/cli/base.py

```diff
@@ -1046,18 +1046,17 @@ def jobspec_create(self, args):
             for keyval in args.setattr:
                 key, val = parse_jobspec_keyval("--setattr", keyval)
 
-                #  If key does not explicitly start with ".", "system."
+                #  If key does not explicitly start with ".", "attributes.", "system."
                 #   or "user.", then "system." is implied. This is a
                 #   meant to be a usability enhancement since almost all
                 #   uses of --setattr will target attributes.system.
                 #
-                if not key.startswith((".", "user.", "system.")):
+                if not key.startswith((".", "attributes.", "user.", "system.")):
                     key = "system." + key
+                elif key.startswith("."):
```

Comment:
I suggest a comment similar to the deleted one to explain why `.` is handled specifically. (This is what tripped me up on a quick review, though perhaps it is more obvious when looking at  code and not the diff.)